Marlin/src/lcd/ultralcd.cpp

```diff
@@ -896,7 +896,21 @@ void MarlinUI::update() {
       if (TERN0(REPRAPWORLD_KEYPAD, handle_keypad()))
         RESET_STATUS_TIMEOUT();
 
-      const float abs_diff = ABS(encoderDiff);
+      float abs_diff = ABS(encoderDiff);
```

Comment:
abs_diff should not be a float.
The only place where abs_diff is used as float is in calculating encoderMovementSteps. But even there it is questionable.
Note that the M3 has no floating point unit.